ampr-ripd: initial package release

K2IE opened the pull request:
Maintainer: Dan Srebnick (K2IE)
Compile tested:  x86 and aarch64
Run tested: x86 and aarch64 23.05.5 and 24.10.0

Description:

This package is intended for licensed radio amateurs who wish to participate on the Net 44 IPIP mesh.  You must be assigned a network via the ARDC portal.  See https://portal.ampr.org/ for more information.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+	DEPENDS:=+kmod-ipip +ip-full
+	TITLE:=Routing daemon for the AMPR network
+	URL:=http://www.yo2loj.ro/hamprojects
+	MAINTAINER:=Dan Srebnick <k2ie@k2ie.net>
```
I think you should set the `PKG_MAINTAINER` instead.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+	$(INSTALL_BIN) ./files/$(PKG_NAME)-init $(1)/etc/init.d/$(PKG_NAME)
+endef
+
+define Package/$(PKG_NAME)/postinst
```
Perhaps use `uci-defaults` script for this?

K2IE replied:
The postinst now just setups some defaults that were approved by Net 44 management.  The export of variables is no longer present nor necessary.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+  amprnet=$$amprnet
+"
+   echo "Creating /etc/ampr-ripd.conf..."
+   echo "# Set to your AMPR allocation
```
This can probably be made clearer with a heredoc.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+TUNNET=$$amprnet/$$amprmask" > /etc/ampr-ripd.conf
+   
+   echo "Installing routing rules..."
+   r=`uci add network rule`
```
Backtick notation is considered legacy: https://www.shellcheck.net/wiki/SC2006

K2IE replied:
That is correct.  For unknown reasons, the $(...) notation totally breaks the ability to add rules and insert the related configuration items.

GeorgeSapkin replied:
You probably just needed to escape the `$` like `$$(...)`. This is not going to be an issue if you use anything other than the `postinst` hook.

GeorgeSapkin commented:
Make sure you're following the [contributing guidelines](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md), wrt commit naming and signing.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,33 @@
+fi
+
+start() {
+   ip tunnel change ttl 64 mode ipip tunl0
```
Where is `tunl0` coming from? And should it be a configuration option?

GeorgeSapkin replied:
I still feel like `tunl0`, that is used in multiple places, is unclear, and can be named and documented better. What if there's already a `tunl0`?

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ampr-ripd
+PKG_VERSION:=2.4
```
It looks like 2.4.1 is the latest version. Any reason for not using it?

K2IE replied:
This 2.4.1 is a bugfix for some debug code which is not used in the package build.  As my package has been in use for a number of years at my site, it has not been tested with the fix that won't be relevant for this use case.

GeorgeSapkin replied:
I'm not sure that justifies including a knowingly-outdated version. If it's not relevant, why not us the latest version either way? This looks like it will result in somebody noticing the package is outdated and making a PR with a version bump.

K2IE replied:
If not using 2.4.1 is considered to be a show stopper by the OpenWRT team, I will withdraw the pull request.  2.4 has been field tested for years, so I am certain of the stability.  I would want to test extensively with 2.4.1.

Please advise whether I shouild withdraw the request or continue on fixing other issues raised in the comments.

I am a member of the Technical Advisory Committee for ARDC (who administers Net 44) and was hoping to make it easier for legitimate users of Net 44 to use OpenWRT as a platform, so I think that the benefits of going ahead with 2.4 outweigh any perceived disadvantages.

GeorgeSapkin replied:
I'm just giving you my opinion about how it looks like from the outside and what might inevitably happen (wrt somebody bumping the patch version). If this is such an important point, perhaps add a comment why it's not at the latest version, and what the potential consequences of updating to it might be.

K2IE replied:
Staying with 2.4 pending a look into why 2.4.1 does not build.

kb2ear replied:
Tested on mpc85xx/p1010 target (Sophos RED 15w Rev.1, OpenWrt 23.05.2 r23630-842932a63d / LuCI openwrt-23.05 branch git-23.306.39416-c86c256
).  Installed fine, no issues.  As a licensed amateur radio operator (KB2EAR) this would be a benefit to the community.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+define Package/$(PKG_NAME)/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/$(PKG_NAME) $(1)/usr/sbin
+	$(INSTALL_DIR) $(1)/etc
```
This looks redundant.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+if [ -z "$${IPKG_INSTROOT}" ]; then
+   echo "Installing with values:
+
+  amprhost=$$amprhost
+  amprmask=$$amprmask
+  amprnet=$$amprnet
```
I'm not sure I understand what's happening here. Is this assigning variables to themselves? Where are these coming from?

K2IE replied:
Prior to installation, these variable needs to be exported.  I am unaware of a method to pass these parameters to the install.

GeorgeSapkin replied:
How would that work when installing a package from a web UI, for example? Perhaps consider setting sane, configurable defaults (i.e. define a default config file)? That when changed will be picked up by the startup script.

K2IE replied:
There are NO "sane" configurable defaults of which I am aware.  This would have to be investigated.

K2IE replied:
I am not certain whether this is feasible.  The values are required to set the various needed uci network and firewall rules.  With no values set before running postinst, I don't believe that configuration could be automated and that is the intent.  I realize that this is not standard, but this package would be a special use case for authorized Net 44 users.

Here is an example of what it looks like at my site:

```
root@openwrt-cm4:~# opkg install ampr-ripd_2.4-1_aarch64_cortex-a72.ipk 
Installing ampr-ripd (2.4-1) to root...
Configuring ampr-ripd.
Installing with values:

  amprhost=44.128.12.33
  amprmask=255.255.255.240
  amprnet=44.128.12.32

Creating /etc/ampr-ripd.conf...
Installing routing rules...
Installing network interfaces...
Installing firewall zones...
Installing firewall rules...
```

With no values, there is little to do.

GeorgeSapkin replied:
You could configure most of the things on install (same code as below without the variable bits), disable the interfaces, and only reconfigure and re-enable them with the missing bits of info from the config on service start.

K2IE replied:
Configuration moved to the initscript.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+PKG_MD5SUM:=5429401ab6f8f793448bef11029d7ead
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tgz
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
This is already default.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+PKG_VERSION:=2.4
+PKG_RELEASE:=1
+PKG_SOURCE_URL:=http://www.yo2loj.ro/hamprojects
+PKG_MD5SUM:=5429401ab6f8f793448bef11029d7ead
```
This looks like to me as md5, which was deprecated here. You should switch to PKG_HASH, which uses sha256

BKPepe replied:
Use https, please.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+	SUBMENU:=Routing and Redirection
+	DEPENDS:=+kmod-ipip +ip-full
+	TITLE:=Routing daemon for the AMPR network
+	URL:=http://www.yo2loj.ro/hamprojects
```
Https here as well

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/$(PKG_NAME)    
```
Dont reuse PKG_NAME variable here and also below.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,181 @@
+	$(INSTALL_BIN) ./files/$(PKG_NAME)-init $(1)/etc/init.d/$(PKG_NAME)
+endef
+
+define Package/$(PKG_NAME)/postinst
```
It looks like you are trying using something, which is not intented to do. This is no-go from my side. You should look to much more suitable solution

K2IE replied:
To clarify, are you stating here that you don't approve of using the postinst script to configure the various network/firewall settings?  One of my major objectives in providing this package is to achieve ease of configuration.  If that is the case, I ought to probably withdraw the pull request and reconsider my overall approach.

BKPepe commented on net/ampr-ripd/README.md:
```diff
@@ -0,0 +1,28 @@
+This project can be used with the OpenWRT SDK to generate a package for ampr-ripd.  It is intended for use only by licensed amateur radio operators.
```
```suggestion
This project can be used with the OpenWrt SDK to generate a package for ampr-ripd.  It is intended for use only by licensed amateur radio operators.
```

BKPepe commented on net/ampr-ripd/patches/000-Makefile.patch:
```diff
@@ -0,0 +1,17 @@
+--- ampr-ripd-2.4/Makefile	2017-05-29 01:50:48.000000000 -0400
```
Why we should have this patch? Please check OpenWrt guidelines.  It is not possible to disable MAN pages while compiling package?

K2IE replied:
This patch does exactly that, it alters the distributed Makefile to not include man pages.

K2IE replied:
I will attempt to do so, but as this may take time to resolve, will this be a no-go item?

BKPepe commented on net/ampr-ripd/patches/001-ampr-ripd.c.patch:
```diff
@@ -0,0 +1,12 @@
+--- ampr-ripd-2.4/ampr-ripd.c	2017-08-14 21:43:56.000000000 -0400
```
Git header is missing. Is this included in upstream? If not, you should describe what it does and then submit it to upstream.

K2IE replied:
Please elaborate.  Not sure what action to take.

K2IE replied:
The added lines are necessary for the code to compile.  It is not clear why they are not in the original source code.

BKPepe replied:
While using your words and sorry for that, it is not clear, why we should have this patch in our repository, when it is not possible to compile it from the source code. In that case, you should report it to upstream, fill bug report, try to convince them to include it (if possible) and have it included in the next version. We should not carry the patch and keep it in our repository and take care of it. As I said earlier:

This patch is missing, who is the author, commit subject, commit description, etc.

BKPepe reviewed (changes requested):
Just very quick while going through. Also, you should use the latest version, if possible.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,70 @@
+        fi
+     fi
+   done
+   uci commit network
```
Is it enough to set those values and not restart networking?

K2IE replied:
Added message to config process reminding user to restart ampr-ripd and networking.

GeorgeSapkin replied:
You could also figure out which services are affected and need restarting, or restart the whole networking in the script.

K2IE replied:
For the time being, I'm going to leave it as a manual process, giving the user time to examine settings first.

GeorgeSapkin commented on net/ampr-ripd/README.md:
```diff
@@ -0,0 +1,31 @@
+This project can be used with the OpenWrt SDK to generate a package for
+ampr-ripd.  It is intended for use only by licensed amateur radio operators.
+Building the OpenWRT package for installation is a separate process.  Detailsi
```
```suggestion
Building the OpenWRT package for installation is a separate process.  Details
```

GeorgeSapkin replied:
Still relevant.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd.conf:
This looks like a config that you're sourcing directly in your service script. Why not use the native uci config facilities for this instead?

GeorgeSapkin reviewed (changes requested):
And again:

> Make sure you're following the [contributing guidelines](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md), wrt commit/PR naming and signing.

You should definitely squash your commits and add a meaningful commit description.

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,70 @@
+
+start() {
+   if [ -f /etc/ampr-ripd.conf ]; then
+      if [ -z $(grep -c ^TUNNET=44.127.254/255.255.255.0 /etc/ampr-ripd.conf) ]; then
```
Handled in next commit.

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,70 @@
+   amprnet=$3
+   echo "Configuring ampr-ripd with values:
+
+amprhost=$amprhost
```
Addressed in next commit.

K2IE commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,171 @@
+	DEPENDS:=+kmod-ipip +ip-full
+	TITLE:=Routing daemon for the AMPR network
+	URL:=https://www.yo2loj.ro/hamprojects
+	PKG_MAINTAINER:=Dan Srebnick <k2ie@k2ie.net>
```
Addressed in next commit.

K2IE commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,171 @@
+	$(INSTALL_BIN) ./files/ampr-ripd-init $(1)/etc/init.d/ampr-ripd
+endef
+
+define Package/ampr-ripd/postinst
```
@GeorgeSapkin  In thinking about using uci-defaults, it is my understand that this runs on boot.  That would introduce an unncessary reboot prior to the ability to use the package.   Am I correct?

K2IE replied:
Now using uci-defauts.

GeorgeSapkin reviewed (changes requested):
I'm repeating myself here, but please address the previous comments regarding:

- using the `uci-defaults` (see other projects)
- storing package configs using UCI and not random config file in `/etc`
- commit squashing, formatting, description, etc.
- PR name

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,82 @@
+   if [ $(uci get network.amprlan.ipaddr) = "44.127.254.254" ] | \
+      [ $(uci get network.amprwan.ipaddr) = "44.127.254.254" ]; then
+
+cat <<EOF
+ampr-ripd is not fully configured.
+You must run /etc/init.d/ampr-ripd configure.
+EOF
```
Will be addressed in the next   commit.

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,82 @@
+EXTRA_HELP="        configure       Configure service parameters"
+
+start() {
+   . /usr/lib/ampr-ripd/defaults.sh
```
I took this approach becase I see that scripts in /usrlib/unbound are being used for configuration.  If good enough for unbound, can you please explain why this approach is unsuitable?

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,82 @@
+
+start() {
+   . /usr/lib/ampr-ripd/defaults.sh
+   if [ $(uci get network.amprlan.ipaddr) = "44.127.254.254" ] | \
```
Addressed in the next commit.

K2IE commented:
>     * storing package configs using UCI and not random config file in `/etc`

No more /etc/ampr-ripd.conf in next commit.

K2IE commented:
This PR is getting messy, despite my attempt to squish things.  I wonder if we would not all be better off if I close this and reopen with a new PR as a single commit.

I believe I have addressed those change requests that are fixable at the moment and do not rely upon others.

Reviewers, please advise.

GeorgeSapkin commented:
Please don't start another PR with the same changes. Instead please check out some online git tutorials about squashing your commits and force-pushing a branch. There are a few ways to do it, depending on your preferences. Since this is a new package, there should only be one commit in the end: there are 7 at the moment.

K2IE commented:
> Please don't start another PR with the same changes. Instead please check out some online git tutorials about squashing your commits and force-pushing a branch. There are a few ways to do it, depending on your preferences. Since this is a new package, there should only be one commit in the end: there are 7 at the moment.
> 
> I'll review the other changes in the meanwhile.

I seem to have succesfully rebased and pushed everything into a single commit.

GeorgeSapkin commented:
While we're on the subject of commits, as I've mentioned a few times already, please:

> add a meaningful commit description.

And don't forget the 50/72 rule.

K2IE commented:
> While we're on the subject of commits, as I've mentioned a few times already, please:
> 
> > add a meaningful commit description.

Is the initial package release description in adequate in your opinion?  If there are specific criteria that you'd like me to add please be more specific.

> 
> And don't forget the 50/72 rule.

I have Googled this and will endeavor to remember.

GeorgeSapkin commented:
> Is the initial package release description in adequate in your opinion? If there are specific criteria that you'd like me to add please be more specific.

You can put the same blob you have in the README now. This will help people looking through the history later to figure why this package was added and what it did.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,97 @@
+	SECTION:=net
+	CATEGORY:=Network
+	SUBMENU:=Routing and Redirection
+	DEPENDS:=+kmod-ipip +ip-full
```
Do you need `ip-full` or is `ip-tiny` enough?

K2IE replied:
> Do you need `ip-full` or is `ip-tiny` enough?

ip-full is definitely required.

BKPepe replied:
Are you sure? By your init script or anything else? Why is not ip-tiny sufficient?

GeorgeSapkin replied:
I quickly went through the PR and I don't see any `ip` commands that are missing from `ip-tiny`. So which features of `ip-full` are required?

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+
+   uci set network.amprlan.ipaddr=$amprhost
+   uci set network.amprlan.netmask=$amprmask
+   uci set network.amprlan.tunnet=$amprnet/$amprmask
```
`tennet` doesn't seem like a standard network field. Consider a separate UCI-based config file. I've been suggesting this for a while now.

GeorgeSapkin replied:
Still unclear.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+   uci set network.amprwan.ipaddr=$amprhost
+   uci set network.amprwan.netmask=$amprmask
+   for i in $(seq 99 -1 0); do
+     if [ $(uci -q get network.@rule[$i].src) ]; then
```
I don't think a subshell is necessary here:

```suggestion
     if uci -q get network.@rule[$i].src; then
```

This might be the same for some other `if` checks that have a subshell now, e.g. in `postrm`.

GeorgeSapkin reviewed (changes requested):
Consider running all the scripts through `shellcheck` and figure which advice applies to BusyBox ash.

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+   ip tunnel change ttl 64 mode ipip tunl0
+   ip link set dev tunl0 up
+   ifconfig tunl0 mtu 1480
+   TUNNET=$(uci -q get network.amprlan.tunnet)
```
Cosmetic, but because TUNNET is used as a parameter to actually run the process and I want to make sure it stood out.

K2IE commented:
@BKPepe @GeorgeSapkin I have a plan to deal with the patches, but this is going to take time and involve the upstream and possibly others.  It could ultimately result in a fork in order to satisfy the requirement to rid the package of the patches.  With this in mind, can we not let this minor matter get in the way of progress and move things along with my commitment that it will be dealt with in the near future?

thom24 commented on net/ampr-ripd/files/99-ampr-ripd:
```diff
@@ -0,0 +1,91 @@
+uci set network.amprwan.ipaddr='44.127.254.254'
+uci set network.amprwan.netmask='255.255.255.0'
+uci commit network
+
+echo Installing default firewall zones...
+z=$(uci add firewall zone)
+uci set firewall.$z.name='amprlan'
+uci set firewall.$z.network='amprlan'
+z=$(uci add firewall zone)
+uci set firewall.$z.name='amprwan'
+uci set firewall.$z.network='amprwan'
+z=$(uci add firewall forwarding)
+uci set firewall.$z.src='amprlan'
+uci set firewall.$z.dest='amprwan'
+uci commit firewall
+
+echo Installing default firewall rules...
+f=$(uci add firewall rule)
+uci set firewall.$f.name='ipip'
+uci set firewall.$f.proto='ipencap'
+uci set firewall.$f.src='wan'
+uci set firewall.$f.target='ACCEPT'
+uci set firewall.$f.family='ipv4'
+uci set firewall.$f.icmp_type='echo-request'
+f=$(uci add firewall rule)
+uci set firewall.$f.name='Net 44 ICMP Echo Request'
+uci set firewall.$f.proto='icmp'
+uci set firewall.$f.src='amprwan'
+uci set firewall.$f.dest='amprlan'
+uci set firewall.$f.target='ACCEPT'
+uci set firewall.$f.family='ipv4'
+uci set firewall.$f.icmp_type='echo-request'
+f=$(uci add firewall rule)
+uci set firewall.$f.name='Net 44 Router ICMP'
+uci set firewall.$f.proto='icmp'
+uci set firewall.$f.src='amprwan'
+uci set firewall.$f.target='ACCEPT'
+uci set firewall.$f.family='ipv4'
+uci set firewall.$f.icmp_type='echo-request'
+uci commit firewall
```
use uci batch

BKPepe replied:
@K2IE dont mark conversations as resolved, when it is not resolved.

GeorgeSapkin commented:
As this is quite an unusual package, seeing how it integrates with the networking stack, and requires a radio license to operate, it would make sense to me that more end users (i.e. besides yourself) would chime in with their test results.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,97 @@
+	$(INSTALL_DATA) ./files/99-ampr-ripd $(1)/etc/uci-defaults/99-ampr-ripd
+endef
+
+define Package/ampr-ripd/postrm
```
With opkg `postrm` is called _before_ an upgrade with `PKG_UPGRADE=1`. So if you don't want all of the configs to be deleted on each upgrade, you should take this into account, e.g.:

```bash
[ "$${PKG_UPGRADE}" = 1 ] && exit 0
```

BKPepe replied:
Those should be marked as conffiles. :)

kb2ear commented:
Tested on mpc85xx/p1010 target (Sophos RED 15w Rev.1, OpenWrt 23.05.2 r23630-842932a63d / LuCI openwrt-23.05 branch git-23.306.39416-c86c256
).  Installed fine, no issues.  As a licensed amateur radio operator (KB2EAR) this would be a benefit to the community.

K2IE commented:
Now on 2.4.1 and no more patches.

GeorgeSapkin commented:
There are a few comments, that are marked as resolved, while they aren't. Can you go through them and check once more?

K2IE commented:
@GeorgeSapkin I have gone through all resolved comments and to the best of my knowledge and belief, they have been addressed.  If there is a particular comment or comments on which you disagree, please point them out.  Note that many comments are outdated because of changes.  I am eager to get this pushed and move on to other things.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+   amprmask=$2
+   amprnet=$3
+
+	cat <<-EOF
```
Still relevant. And applies to all the scripts.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+   uci set network.amprwan.netmask=$amprmask
+   for i in $(seq 99 -1 0); do
+     if [ $(uci -q get network.@rule[$i].src) ]; then
+        src=$(uci get network.@rule[$i].src)
```
Still unclear.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,98 @@
+define Package/ampr-ripd/postrm
+#!/bin/sh
+[ "$${PKG_UPGRADE}" = 1 ] && exit 0
+if [ -z "$${IPKG_INSTROOT}" ]; then
```
You could probably do a similar thing as with `PKG_UPDATE` to reduce indentation:

```suggestion
[ -z "$${IPKG_INSTROOT}" ] || exit 0
```

GeorgeSapkin reviewed (changes requested):
I commented on all the items that are still unresolved or unclear. Some of them are only visible in the main conversation tab, and not in changes. Particularly the `tennet` config option and the `tunl0` device names are of note. And the mix of tabs and spaces in the service script.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,98 @@
+PKG_NAME:=ampr-ripd
+PKG_VERSION:=2.4.1
+PKG_RELEASE:=1
+PKG_SOURCE_URL:=https://k2ie.net/ftp
```
I dont think that we should use your website to download the latest version. It is different than the official site https://www.yo2loj.ro/hamprojects. We dont know what was changed, etc. This could be also security flaw. 

Why have you decided to use this approach? Why not download Git sources from your fork to see what have you done? Anyway, downloading from the official site should be the way.

BKPepe commented:
What happened with those patches, which you have earlier? Dont tell me that you include it in your mistaken tarball without saying anything. :) I dont see tem also included or create issue here: https://git.ampr.org/yo2loj/ampr-ripd that you tried to work with upstream.

K2IE commented:
> What happened with those patches, which you have earlier? Dont tell me that you include it in your mistaken tarball without saying anything. :) I dont see tem also included or create issue here: https://git.ampr.org/yo2loj/ampr-ripd that you tried to work with upstream.

I have integrated the changes into the source as the quickest way to solve the problem.  The patches originally came from someone who submitted them long ago to Marius as the way to get this program to compile for use with OpenWrt (but outside a package).

I have been running this package for a number of years and can no longer find a link to the original source of the now removed packages, which simply rid us of the manpage and add two defines.

Is this a no-go?

K2IE commented:
> I commented on all the items that are still unresolved or unclear. Some of them are only visible in the main conversation tab, and not in changes. Particularly the `tennet` config option and the `tunl0` device names are of note. And the mix of tabs and spaces in the service script.

I have responded to all these items but will repeat hear for clarity:

tunl0 -- I have stated that this COULD be a configuration item in the future but is extremely unlikely to be needed by anyone running ampr-ripd for Net 44 IPIP mesh.  This is a legacy choice and stems from all the existing documentation on using ampr-ripd with OpenWrt outside of a package.  Please see https://wiki.ampr.org/wiki/Setting_up_a_gateway_on_OpenWRT.  This is the mess that I am trying to save folks from, hacking up their OpenWrt configs.

tennet -- If you are referring to tunnet, exactly what is the objection now.? I recall you did not like it upper cased, we discussed saving the value using uci...please be more specific.  Often the "unclear" comments are not clear themselves as to what is being objected to.

indentation -- I again repeat that the indentation was consistent prior to change a bunch of echos to heredocs.  This made indentation necessary where spaces had been used.  The requirement at the time was not used heredoc, but be aware that all your space indentation will now have to be changed to tabs and you must reformat the entire script.  That would have been specific.  Is this a no-go item?

GeorgeSapkin commented:
> tennet -- If you are referring to tunnet, exactly what is the objection now.? I recall you did not like it upper cased, we discussed saving the value using uci...please be more specific. Often the "unclear" comments are not clear themselves as to what is being objected to.

It's `tunnet` of course, I keep fat-fingering it. As I said before, it doesn't seem like a standard network field. So it might be accidentally overridden when corresponding entries are modified through the UI. Consider creating a separate package-specific UCI-based config file, e.g. `/etc/config/amrp-ripd`, where you can store this value or `net` and `mask` separately if that makes sense.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+
+start() {
+	default_addr="44.127.254.254"
+	if [ $(uci get network.amprlan.ipaddr) = "$default_addr" ] | \
+	   [ $(uci get network.amprwan.ipaddr) = "$default_addr" ]; then
```
I think you meant _or_ and not pipe:
```suggestion
	if [ $(uci get network.amprlan.ipaddr) = "$default_addr" ] || \
	   [ $(uci get network.amprwan.ipaddr) = "$default_addr" ]; then
```

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,77 @@
+		if [ ! -z $(uci -q get network.@rule[$i].src) ]; then
+			src=$(uci get network.@rule[$i].src)
+			lookup=$(uci get network.@rule[$i].lookup)
+		        if [ "$lookup" = "44" ]; then
```
Again, tabs and spaces.

GeorgeSapkin commented:
> I have responded to all these items but will repeat hear for clarity:

There are still items that you have not responded to, like the unused `src` variable.

K2IE commented:
> > I have responded to all these items but will repeat hear for clarity:
> 
> There are still items that you have not responded to, like the unused `src` variable.

Done.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,78 @@
+}
+
+configure() {
+	if [ -z $1 ] | [ -z $2 ] | [ -z $3 ]; then
```
```suggestion
	if [ -z $1 ] || [ -z $2 ] || [ -z $3 ]; then
```

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,78 @@
+	amprnet=$amprnet
+	EOF
+
+	[ -f /etc/config/ampr-ripd ] || touch /etc/config/ampr-ripd
```
Instead of creating a new config file, you should include an example config that describes all the possible options. In this case, just one, so far.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,78 @@
+	uci set network.amprwan.netmask=$amprmask
+	for i in $(seq 99 -1 0); do
+		if [ ! -z $(uci -q get network.@rule[$i].src) ]; then
+			lookup=$(uci get network.@rule[$i].lookup)
```
Since all the rules have known `src` and `dest`, can't you use those for looking up values? This is the same issue as the one with having the custom `tunnet` option previously.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+	$(INSTALL_DATA) ./files/99-ampr-ripd $(1)/etc/uci-defaults/99-ampr-ripd
+	$(INSTALL_DIR) $(1)/etc/config
+	$(INSTALL_DATA) ./files/ampr-ripd-config $(1)/etc/config/ampr-ripd
+
```
```suggestion
```

GeorgeSapkin replied:
There's still this empty line.

GeorgeSapkin commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,78 @@
+	ip link set dev tunl0 up
+	ifconfig tunl0 mtu 1480
+	tunnet=$(uci -q get ampr-ripd.network.tunnet)
+	/usr/sbin/ampr-ripd -s -r -t 44 -i tunl0 -a $tunnet
```
```suggestion
	/usr/sbin/ampr-ripd -s -r -t 44 -i tunl0 -a "$tunnet"
```

Also, and it's not critical, but consider using procd: https://openwrt.org/docs/guide-developer/procd-init-script-example

GeorgeSapkin reviewed (commented):
I'd probably run all the scripts through shellcheck and fix all the warnings that apply.

You also could add a bit more comments to the bigger scripts.

Other than that, the source/patch situation is the only concern now.

K2IE commented:
> Just very quick while going through. Also, you should use the latest version, if possible.

This is now addressed.  I have worked with the upstream developer to address the various OpenWrt related issues raised here.

K2IE commented:
> I'd probably run all the scripts through shellcheck and fix all the warnings that apply.
> 
> You also could add a bit more comments to the bigger scripts.
> 
> Other than that, the source/patch situation is the only concern now.

Source/patch situation resolved.

GeorgeSapkin commented on net/ampr-ripd/README.md:
````diff
@@ -0,0 +1,40 @@
+
+```
+opkg update
+opkg install ampr-ripd_2.4-r1_{your architecture}.ipk
````
@K2IE I think you should address this: it looks confusing to me as well.

BKPepe replied:
What about this one as well @K2IE ?

GeorgeSapkin commented:
Perhaps both you - @K2IE, and @kb2ear can re-test this, including upgrading the package (i.e. force-installing the same version over the existing one) and uninstalling, to make sure nothing falls apart in these scenarios. Testing against the snapshot version would be useful too.

K2IE commented:
> Perhaps both you - @K2IE, and @kb2ear can re-test this, including upgrading the package (i.e. force-installing the same version over the existing one) and uninstalling, to make sure nothing falls apart in these scenarios. Testing against the snapshot version would be useful too.

--force-reinstall and --force-overwrite tested, works as expected.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+	SECTION:=net
+	CATEGORY:=Network
+	SUBMENU:=Routing and Redirection
+	DEPENDS:=+kmod-ipip +ip-tiny
```
Since you switched the dependency from `ip-full` to `ip-tiny` as per previous discussion, you can specify it as `+ip` instead, as mentioned before, so whichever version a user has installed out of those two will be fine.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+	$(INSTALL_DIR) $(1)/etc/uci-defaults
+	$(INSTALL_DATA) ./files/99-ampr-ripd $(1)/etc/uci-defaults/99-ampr-ripd
+	$(INSTALL_DIR) $(1)/etc/config
+	$(INSTALL_DATA) ./files/ampr-ripd-config $(1)/etc/config/ampr-ripd
```
This file should be marked as conffile. Shouldnt it be?

BKPepe replied:
Still applies.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+
+endef
+
+$(eval $(call BuildPackage,$(PKG_NAME)))
```
Why using here PKG_NAME variable here? :) No need.

```suggestion
$(eval $(call BuildPackage,ampr-ripd))
```

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+if [ -f /etc/config/ampr-ripd ]; then
+   uci del ampr-ripd.tunnet
+   uci commit ampr-ripd
+   rm -f /etc/config/ampr-ripd
```
Why?

BKPepe replied:
Well, waiting for respond since I am marked again for review.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tgz
+PKG_LICENSE:=GPL-2.0-only
+
```
```suggestion
PKG_LICENSE_FILES:=COPYING

```

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ampr-ripd
+PKG_VERSION:=2.4.2
+PKG_RELEASE:=1
+PKG_SOURCE_URL:=https://yo2loj.ro/hamprojects
+PKG_HASH:=e4635bd0f88c1f2b0777e948a00d0470aa97254ec4b0b8fd75c79d109995a350
+PKG_MAINTAINER:=Dan Srebnick <k2ie@k2ie.net>
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tgz
```
```suggestion
PKG_NAME:=ampr-ripd
PKG_VERSION:=2.4.2
PKG_RELEASE:=1

PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tgz
PKG_SOURCE_URL:=https://yo2loj.ro/hamprojects
PKG_HASH:=e4635bd0f88c1f2b0777e948a00d0470aa97254ec4b0b8fd75c79d109995a350

PKG_MAINTAINER:=Dan Srebnick <k2ie@k2ie.net>
```

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/ampr-ripd
+	SECTION:=net
+	CATEGORY:=Network
+	SUBMENU:=Routing and Redirection
+	DEPENDS:=+kmod-ipip +ip-tiny
+	TITLE:=Routing daemon for the AMPR network
+	URL:=https://www.yo2loj.ro/hamprojects
+endef
+
+define Package/ampr-ripd/description
+	Routing daemon written in C similar to Hessu's rip44d including optional resending of RIPv2 broadcasts for router injection.
+endef
```
```suggestion
define Package/ampr-ripd
  SECTION:=net
  CATEGORY:=Network
  SUBMENU:=Routing and Redirection
  DEPENDS:=+kmod-ipip +ip-tiny
  TITLE:=Routing daemon for the AMPR network
  URL:=https://www.yo2loj.ro/hamprojects
endef

define Package/ampr-ripd/description
  Routing daemon written in C similar to Hessu's rip44d including optional resending of RIPv2 broadcasts for router injection.
endef
```

1715173329 replied:
the description should be wrapped for every ~70 chars

BKPepe commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,87 @@
+		[ "$(uci get network.amprwan.ipaddr)" = "$default_addr" ]; then
+
+		cat <<-EOF
+		ampr-ripd is not fully configured.
+		You must run /etc/init.d/ampr-ripd configure.
```
This could be done by enabled option in config.

BKPepe replied:
:)

BKPepe commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,87 @@
+EXTRA_HELP="        configure       Configure service parameters"
+
+start() {
+	default_addr="44.127.254.254"
```
I wonder. This IP can be configured, right? Why it is not in config file and you do it in init script?
Also, this IP address is created by uci-defaults.

I find this confusing, though.

BKPepe replied:
Kind reminder here as well. :)

BKPepe commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,87 @@
+	if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
+		cat <<-EOF
+
+		Usage: /etc/init.d/ampr-ripd configure [amprhost] [amprmask] [amprnet]
+
+		       amprhost is the 44 net address assigned to your OpenWrt host
+		       amprmask is the full netmask of your AMPR assigned network
+		       amprnet is the network number of your AMPR assigned network 
+
+		   Eg. /etc/init.d/ampr-ripd configure 44.127.254.1 255.255.255.0 44.127.254.0
```
Thinking about this. Shouldnt the binary help you with that? I think you are trying reinvent a wheel here, which leads to be things really confusing and complicated. They don't need to be like that.

GeorgeSapkin replied:
I'm not sure I follow: what should the binary help with in this case?

BKPepe replied:
I mean the resulting binary on the router itself. It should not start if it is not configured or also it should provide ways how to provide it, if you want to have it more user-friendly.

E.g. /usr/sbin/ampr-ripd configure or something like it. 

But still, in my point of view, this needs to be done differently, because it does so *magical* things to do network, which is not so good. I can not imagine situation that I will install this package and it tweaks my network so much/so badly (adding default rules, network interfaces, firewall zones, etc) and then removing it. What if something goes bad and that it could not be removed, added?

It is done so complicated. 🤷

GeorgeSapkin replied:
I see where you're coming from. This seems like quite a specialized package, so perhaps potential users won't mind.

As an aside, have you seen a similar approach to changing multiple system-wide settings in the open Tailscale PR? I don't want to link it here to create noise.

BKPepe replied:
Unfortunately, not yet.

BKPepe commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,109 @@
+PKG_SOURCE_URL:=https://yo2loj.ro/hamprojects
+PKG_HASH:=e4635bd0f88c1f2b0777e948a00d0470aa97254ec4b0b8fd75c79d109995a350
+
+PKG_MAINTAINER:=Dan Srebnick <k2ie@k2ie.net>
+
+PKG_LICENSE:=GPL-2.0-only
+PKG_LICENSE_FILES:=COPYING
```
```suggestion
PKG_MAINTAINER:=Dan Srebnick <k2ie@k2ie.net>
PKG_LICENSE:=GPL-2.0-only
PKG_LICENSE_FILES:=COPYING
```

BKPepe reviewed (changes requested):
As I was requested again for review, please check previous comments. I tried to respond to some of them to remind it for you.

K2IE commented:
@BKPepe I seem to be having a problem with passing the formalities check.

    Signed-off-by: Dan Srebnick <dan@islenet.com>

The signed-off-by line being used seems to be in conformance with requirements.

$git config --global user.name
Dan Srebnick

$git config --global user.email
dan@islenet.com

According to the formalities log:

Run source .github/workflows/ci_helpers.sh
: === Checking commit 'e7bbd3ba8194e3b02698792015a8f5f79cba82c0'
: Author name (K2IE) need to be your real name 'firstname lastname'
: Commit subject line seems ok (ampr-ripd: Initial package release)
: Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: K2IE <dan@islenet.com>')
Error: Process completed with exit code 1.

I must be missing something here, guidance appreciated.

GeorgeSapkin commented:
> : Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: K2IE [dan@islenet.com](mailto:dan@islenet.com)')

It's telling you here that commit author is `From: K2IE <dan@islenet.com>` which doesn't match the sign off.

K2IE commented:
> > : Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: K2IE [dan@islenet.com](mailto:dan@islenet.com)')
> 
> It's telling you here that commit author is `K2IE <dan@islenet.com>` which doesn't match the sign off.

@GeorgeSapkin All documentation re signoff that I have read states "Firstname Lastname", so to confirm, using K2IE here is good even though it does not fulfill the documented requirements...correct?

GeorgeSapkin commented:
> All documentation re signoff that I have read states "Firstname Lastname", so to confirm, using K2IE here is good even though it does not fulfill the documented requirements...correct?

As mentioned in the [contributing guidelines](https://github.com/openwrt/packages/blob/cffbe8b32a26efeaae0f0d8ce896756da77e16e2/CONTRIBUTING.md#commits-in-your-pull-requests-should) (emphasis mine):

> **Author and sign-off must match** and be a **real name or known identity** and a real email address. GitHub private email addresses will not be accepted.

And that's what this formalities check is telling you.

You can edit the author for the latest commit using:

```bash
git commit --amend --author="Dan Srebnick <dan@islenet.com>"
```

K2IE commented:
> > All documentation re signoff that I have read states "Firstname Lastname", so to confirm, using K2IE here is good even though it does not fulfill the documented requirements...correct?
> 
> As mentioned in the [contributing guidelines](https://github.com/openwrt/packages/blob/cffbe8b32a26efeaae0f0d8ce896756da77e16e2/CONTRIBUTING.md#commits-in-your-pull-requests-should) (emphasis mine):
> 
> > **Author and sign-off must match** and be a **real name or known identity** and a real email address. GitHub private email addresses will not be accepted.
> 
> And that's what this formalities check is telling you.
> 
> You can edit the author for the latest commit using:
> 
> ```shell
> git commit --amend --author="Dan Srebnick <dan@islenet.com>"
> ```

Must the email address be the one associated with github or could I use a preferred address?

git commit --amend --author="Dan Srebnick <k2ie@k2ie.net>"

GeorgeSapkin commented:
> Must the email address be the one associated with github or could I use a preferred address?

Any real address is fine as long as it matches in author and sign-off.

K2IE commented:
@BKPepe It appears that a number of comment replies which I thought you had seen weeks ago were still in pending.  That has been remedied.

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,87 @@
+	uci set network.amprlan.netmask="$amprmask"
+	uci set network.amprwan.ipaddr="$amprhost"
+	uci set network.amprwan.netmask="$amprmask"
+	for i in $(seq 0 -1 -99); do
```
This is intentional.  The negative indices work as intended.  For example, -1 returns the last entry in the list.  -2 returns the next to last entry in the list.

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,87 @@
+	uci set network.amprwan.ipaddr="$amprhost"
+	uci set network.amprwan.netmask="$amprmask"
+	for i in $(seq 0 -1 -99); do
+		if [ ! -z $(uci -q get network.@rule[$i].src) ] && \
```
In this usage, the suggested quotes are not necessary because we don't care about the contents of the configuration item, rather we want to know if it exists.

K2IE commented on net/ampr-ripd/README.md:
```diff
@@ -0,0 +1,40 @@
+
+The target system must have the openwrt_kmods feed listed in
+/etc/opkg/distfeeds.conf.  If not, you may manually add the feed.
+Alternatively, your may wish to use the firmware-selector to build an
```
Typo fixed.

K2IE commented:
I see that a commit was done.  I also see that "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."  Is there a further todo to add this to a branch?  I can certify that it has been test and works in 24.10.x as well as 23.05.x.

GeorgeSapkin reviewed (changes requested):
You can't have merge commits in your PR, so you need to squash them.

I'm still not sure what value the README adds. If users can't install a package, they're not going to find it buried in the git repo. The rest can be part of CLI help.

K2IE commented:
> You can't have merge commits in your PR, so you need to squash them.
> 
> I'm still not sure what value the README adds. If users can't install a package, they're not going to find it buried in the git repo. The rest can be part of CLI help.

README.md removed.   Merge commits squashed.

K2IE commented:
@BKPepe I fixed email to match which seems to be wanted by the test.  

 Signed-off-by: K2IE <k2ie@k2ie.net>

 At your convenience, please re-approve workflows.  Apologies for the inconvenience.

K2IE commented:
> @BKPepe I fixed email to match which seems to be wanted by the test.
> 
> Signed-off-by: K2IE [k2ie@k2ie.net](mailto:k2ie@k2ie.net)
> 
> At your convenience, please re-approve workflows. Apologies for the inconvenience.

Failed!  For some reason, the script seems to want:

`: Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Dan Srebnick <k2ie@k2ie.net>')`

I have adjusted the commit as follows:

```
Author: Dan Srebnick <k2ie@k2ie.net>
Date:   Thu Apr 3 16:15:33 2025 -0400

    ampr-ripd: Initial package release
    
    This project can be used with the OpenWrt SDK to generate a package
    for ampr-ripd.  It is intended for use only by licensed amateur radio
    operators.  ampr-ripd is a modified RIPv2 listener that provides
    route information for ARDC IPIP Mesh tunnels.
    
    Signed-off-by: Dan Srebnick <k2ie@k2ie.net>
```

Ready for retry.

K2IE commented:
@BKPepe any further changes requested?

K2IE commented:
@BKPepe @GeorgeSapkin sorry to bother you but I would like to understand what is needed to move ahead or if this effort was for naught.

kb2ear commented:
Will this be merged?  I and my fellow amateur radio operators would use this functionality.

GeorgeSapkin commented:
@K2IE I didn't have any objections last I checked. However I don't have a say about what gets merged.

K2IE commented:
@BKPepe @GeorgeSapkin this has been pending for almost 2 months.  I would like to understand next steps.

GeorgeSapkin commented:
I will re-review, build the package locally and see if this still makes sense. And if all is well, merge it.

GeorgeSapkin commented:
OK, so I built and tested the package in QEMU, and it install and starts fine after doing the suggested config. On uninstall all the rules were deleted from firewall and network. I didn't see anything unusual come up in logs. I can't test it beyond that.

This came up in previous discussions already, and I still don't think it's customary (as in no other package does that) to delete the main config `/etc/config/ampr-ripd` in `postrm` on uninstall though. I'd say this should go.

And for future updates, I think the main process `ampr-ripd` should be managed by [procd](https://openwrt.org/docs/guide-developer/procd-init-scripts). You don't need to change anything about this now.

GeorgeSapkin commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,107 @@
+uci del network.amprlan
+uci commit network
+
+if [ -f /etc/config/ampr-ripd ]; then
+   uci del ampr-ripd.tunnet
+   uci commit ampr-ripd
+fi
```
Sorry, I wasn't explicit in my last comment, I should have mentioned this whole section that's modifying and deleting (this part is gone already) the main config is unusual. I have retested the latest changes and it doesn't work on a modified config either way.

If you remove this I think we can merge the PR.

K2IE replied:
Would you accept:

if [ -f /etc/config/ampr-ripd ]; then
   uci set ampr-ripd.network.tunnet='44.127.254.0/255.255.255.0'
   uci commit ampr-ripd
fi

This will leave the config file in place but replace tunnet with the default unconfigured network.

K2IE replied:
@GeorgeSapkin upon further consideration and testing, I conclude it safe to remove the section you have requested from the Makefile and have pushed the commit.  If we later find that this creates operational concerns I will request a reconsideration.  I hope that you find this satisfactory.

K2IE commented:
@GeorgeSapkin  It was a long road but thank you.  Do I now submit a PR to the 24.10 branch to make the package available to the current release?

1715173329 commented on net/ampr-ripd/Makefile:
```diff
@@ -0,0 +1,102 @@
+[ -z "$${IPKG_INSTROOT}" ] || exit 0
+
+echo "Removing firewall rules..."
+for i in $$(seq 99 -1 0); do
```
This really... looks horrible. So you're assuming there are exactly 100 rules, and only 100?

```
uci show firewall | awk "/name='(Net 44 ICMP Echo Request|Net 44 Router ICMP|ipip)'/ {split(\$0, conf, /[.=]/); print conf[2]}"
```

This command should extract all section you want, the same below.

K2IE replied:
See PR #27375

K2IE commented on net/ampr-ripd/files/ampr-ripd-init:
```diff
@@ -0,0 +1,87 @@
+START=95
+STOP=10
+
+EXTRA_COMMANDS="configure"
```
Thank you for pointing this out.  Will be implemented in a future PR.
